src/Catel.Core/Catel.Core.Shared/Logging/LogManager.cs

```diff
@@ -464,26 +462,73 @@ public static ILogListener AddDebugListener(bool ignoreCatelLogging = false)
         }
 
         /// <summary>
+        /// Gets the logger for the specified generic type.
+        /// </summary>
+        /// <typeparam name="T">The type.</typeparam>
+        /// <returns>The <see cref="ILog"/> object for the specified type.</returns>
+        public static ILog GetLogger<T>()
+        {
+            lock (_loggers)
+            {
+                var key = typeof(T).ToString();
+                if (!_loggers.ContainsKey(key))
+                {
+                    var log = new Log(typeof(T));
+                    log.LogMessage += OnLogMessage;
+
+                    _loggers.Add(key, log);
+                }
+
+                return _loggers[key];
+            }
+        }
+
+        /// <summary>
         /// Gets the logger for the specified type.
         /// </summary>
         /// <param name="type">The type.</param>
-        /// <returns></returns>
-        /// <exception cref="ArgumentNullException">The <paramref name="type"/> is <c>null</c>.</exception>
+        /// <returns>The <see cref="ILog"/> object for the specified type.</returns>
+        /// <exception cref="ArgumentNullException">If <paramref name="type"/> is <c>null</c>.</exception>
         public static ILog GetLogger(Type type)
         {
             Argument.IsNotNull("type", type);
 
             lock (_loggers)
             {
-                if (!_loggers.ContainsKey(type))
+                var key = type.ToString();
```

Comment:
FullName instead of ToString? And this part of the code can be outside of the lock.